File: tests/Generator/resource/csharp/complex/Security.cs
using System.Text.Json.Serialization;

namespace TypeAPI.Model;

[JsonPolymorphic(TypeDiscriminatorPropertyName = "type")]
[JsonDerivedType(typeof(SecurityApiKey), typeDiscriminator: "apiKey")]
[JsonDerivedType(typeof(SecurityHttpBasic), typeDiscriminator: "httpBasic")]
[JsonDerivedType(typeof(SecurityHttpBearer), typeDiscriminator: "httpBearer")]
[JsonDerivedType(typeof(SecurityOAuth), typeDiscriminator: "oauth2")]
public abstract class Security
{
    [JsonPropertyName("type")]
    public string? Type { get; set; }

}

File: tests/Generator/resource/rust/complex/security.rs
use serde::{Serialize, Deserialize};
use security_api_key::SecurityApiKey;
use security_http_basic::SecurityHttpBasic;
use security_http_bearer::SecurityHttpBearer;
use security_o_auth::SecurityOAuth;

#[derive(Serialize, Deserialize)]
pub struct Security {
    #[serde(rename = "type")]
    _type: Option<String>,

}

